spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/utils/HiveGravitinoOperationOperator.java

```diff
@@ -0,0 +1,142 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.spark.connector.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.gravitino.exceptions.NoSuchPartitionException;
+import org.apache.gravitino.rel.expressions.literals.Literal;
+import org.apache.gravitino.rel.partitions.Partition;
+import org.apache.gravitino.rel.partitions.Partitions;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.jetbrains.annotations.NotNull;
+
+public class HiveGravitinoOperationOperator {
+
+  private org.apache.gravitino.rel.Table gravitinoTable;
+  private static final String PARTITION_NAME_DELIMITER = "/";
+  private static final String PARTITION_VALUE_DELIMITER = "=";
+
+  public HiveGravitinoOperationOperator(org.apache.gravitino.rel.Table gravitinoTable) {
+    this.gravitinoTable = gravitinoTable;
+  }
+
+  public void createPartition(
+      InternalRow ident, Map<String, String> properties, StructType partitionSchema) {
+    List<String[]> fields = new ArrayList<>();
+    List<Literal<?>> values = new ArrayList<>();
+
+    int numFields = ident.numFields();
+    for (int i = 0; i < numFields; i++) {
+      StructField structField = partitionSchema.apply(i);
+      DataType dataType = structField.dataType();
+      fields.add(new String[] {structField.name()});
+      values.add(SparkPartitionUtils.toGravitinoLiteral(ident, i, dataType));
+    }
+
+    Partition partition =
+        Partitions.identity(
+            null, fields.toArray(new String[0][0]), values.toArray(new Literal[0]), properties);
+
+    gravitinoTable.supportPartitions().addPartition(partition);
+  }
+
+  public boolean dropPartition(InternalRow ident, StructType partitionSchema) {
+    String partitionName = getPartitionName(ident, partitionSchema);
+    return gravitinoTable.supportPartitions().dropPartition(partitionName);
+  }
+
+  public InternalRow[] listPartitionIdentifiers(
+      String[] names, InternalRow ident, StructType partitionSchema) {
+    // Get all partitions
+    if (names != null && names.length == 0) {
+      String[] allPartitions = gravitinoTable.supportPartitions().listPartitionNames();
+      return Arrays.stream(allPartitions)
+          .map(
+              e -> {
+                String[] splits = e.split(PARTITION_NAME_DELIMITER);
```

Comment:
Could you use a func like `toSparkPartition` to encapsulation this logic?  rename  `getPartitionName` to `toHivePartitionName`?